packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,618 @@
+import {
+  Account,
+  AddressString,
+  addressToScriptHash,
+  Block,
+  ClaimTransaction,
+  Hash256String,
+  Input,
+  InvocationTransaction,
+  InvokeSendUnsafeReceiveTransactionOptions,
+  isUInt160,
+  isUInt256,
+  IterOptions,
+  NetworkType,
+  Output,
+  Param,
+  PublicKeyString,
+  RawAction,
+  RawCallReceipt,
+  RawInvocationData,
+  RawInvokeReceipt,
+  ScriptBuilderParam,
+  SourceMaps,
+  Transaction,
+  TransactionOptions,
+  TransactionReceipt,
+  TransactionResult,
+  Transfer,
+  UpdateAccountNameOptions,
+  UserAccount,
+  UserAccountID,
+  UserAccountProvider,
+} from '@neo-one/client-common';
+import { AsyncIterableX } from '@reactivex/ix-es2015-cjs/asynciterable/asynciterablex';
+import BigNumber from 'bignumber.js';
+import BN from 'bn.js';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { clientUtils } from '../clientUtils';
+import { UnknownProviderError } from '../errors';
+import { Hash256 } from '../Hash256';
+import { Argument, AssetInput, AssetOutput, Dapi, DapiNetworks } from './Dapi';
+import { FullTransfer, Provider, UserAccountProviderBase } from './UserAccountProviderBase';
+
+const paramToArgDataType = (value: ScriptBuilderParam | undefined): Argument => {
+  if (value === undefined) {
```

Comment:
`undefined` is supported, this should basically just work like https://github.com/neo-one-suite/neo-one/blob/51c27baa35675740e3ecc9f426395bc127495e65/packages/neo-one-client-common/src/ScriptBuilder.ts#L115

I would also reorder the conditions here to match and put a comment in both places pointing to the other that it needs to be updated if one changes.